src/inmanta/protocol/methods_v2.py

```diff
@@ -1357,3 +1357,93 @@ def set_password(username: str, password: str) -> None:
     :raises NotFound: Raised when the user does not exist
     :raises BadRequest: Raised when server authentication is not enabled
     """
+
+
+@typedmethod(
+    path="/unmanaged/<unmanaged_resource_id>",
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+    varkw=True,
+)
+def unmanaged_resource_create(
+    tid: uuid.UUID, unmanaged_resource_id: str, **kwargs: object  # bypass the type checking for the values
+) -> None:
+    """
+    create a discovered resource.
+    :param tid: The id of the environment this resource belongs to
+    :param unmanaged_resource_id: The id of the unmanaged resource
+    :param values: The values associated with the unmanaged_resource
+    :param **kwargs: The following arguments are supported:
+           values: The values associated with the unmanaged_resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/",
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+)
+def unmanaged_resource_create_batch(tid: uuid.UUID, unmanaged_resources: List[UnmanagedResource]) -> None:
+    """
+    create multiple discovered resource in the DB
+    :param tid: The id of the environment this resource belongs to
+    :param unmanaged_resources: List of UnmanagedResources containing the unmanaged_resource_id and values for each resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/<unmanaged_resource_id>",
+    operation="GET",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def unmanaged_resources_get(tid: uuid.UUID, unmanaged_resource_id: str) -> model.UnmanagedResource:
+    """
+    Get a single discovered resource.
+
+    :param tid: the id of the environment in which to get the unmanaged resource.
+    :param unmanaged_resource_id: The id of the unmanaged resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged",
+    operation="GET",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def unmanaged_resources_get_batch(
+    tid: uuid.UUID,
+    limit: Optional[int] = None,
+    first_id: Optional[model.ResourceVersionIdStr] = None,
```

Comment:
do we need both first_id and start? I think with single orders, this is not required?

(not quite sure)